src/composables/validation.ts

```diff
@@ -0,0 +1,121 @@
+import { clone, getValueByPath, setValueByPath, setValuesAlongPath } from '@grozav/utils';
+import { inject, InjectionKey, provide, ref } from 'vue';
+import { FormKey } from '../components/IForm';
+import { FormGroupKey } from '../components/IForm/components/IFormGroup/mixin';
+import { validate } from '../validation';
+import { useInkline } from './inkline';
+
+const ValidationInjectionKey = Symbol('FormValidation') as InjectionKey<{ schema?: any }>;
+
+export function useValidation(props: {
```

Comment:
Should probably extract the functions form inside here (shouldValidate, setValue, touchValue, etc.) to have them testable individually.